frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx

```diff
@@ -17,117 +17,140 @@ import CustomTextArea from '@Commons/Input/TextArea/CustomTextArea';
 import { faWheatAwn } from '@fortawesome/free-solid-svg-icons';
 import initialFarmDetails from '@Constants/InitialFarmDetails';
 import FieldDetailInterface from 'src/Interface/FieldDetailsInterface';
-import {
-  StyledFarmInfo,
-  StyledTextAreaContainer,
-  StyledButtonGroupContainer,
-  StyledListContainer,
-  StyledListItem,
-} from './FieldsAndSoil.style';
-
-const initialFieldValues = initialFarmDetails.Fields[0];
+import FieldsButtonComponent from './FieldsButtonComponent';
+import FieldsListComponent from './FieldsListComponent';
+import { StyledFarmInfo, StyledTextAreaContainer } from './FieldsAndSoil.style';
+import { StyledButtonGroupContainer } from './FieldsButtonComponent.styles';
 
 const FieldsAndSoilComponent: React.FC<InputModuleProps> = ({ farmDetails, updateFarmDetails }) => {
-  const [fieldsInfo, setFieldsInfo] = useState(farmDetails);
+  // Builds field info inside the field form module.
+  const [, setFieldsInfo] = useState(farmDetails);
+  const [fieldIndex, setFieldIndex] = useState(0);
```

Comment:
This is only being used to set the Field id, right?
If so, would you mind starting the useState with 1 instead of 0?
NMP files do have an id for Fields, but they start from 1 instead of 0 :smile: 
Will make our lifes easier on conversion...

Reply:
Yeah alright. Sounds good

Reply:
if your project wraps up before the end of semester you should write a converter so you can use proper camelCase in your code and it handles the PascalCase NMP is using 😅

Reply:
We can add that as the project's stretch goals. 😄

Reply:
@GDamaso I have to keep it as useState(0)
I get a problem with Formik if I change it into useState(1).
Eitherways, the id will start 1 anyways if I press the addButton 
![image](https://github.com/bcgov/nr-sustainment-capstone-2024/assets/144957476/f6d6e2cd-a742-4389-a162-2715f5dd7e8f)

Reply:
I'm going to have some chart for exporting the data, but the more data manipulation the harder it will get. At this point I'd like us to be as close as possible, unless their structure doesn't make sense for our app. 
Starting it this way will make it easier to implement now, and hopefully update to whatever cRaZyCasIng we want in the future.